server.go

```diff
@@ -490,7 +499,9 @@ func (s *server) Start() error {
 	if err := s.cc.chainNotifier.Start(); err != nil {
 		return err
 	}
-
+	if err := s.sphinx.Start(); err != nil {
+		return err
+	}
```

Comment:
Rather than extend the `OnionProcessor` interfaces with `Start/Stop` methods. I think the server can handle the lifetime of its operation within it's _own_ `Start/Stop` methods.